users/vieting/experiments/switchboard/ctc/feat/network_helpers/perturbation.py

```diff
@@ -1,31 +1,27 @@
 from typing import List, Dict, Optional, Any
 
-
 class PerturbationFactor:
     """
     Class to wrap perturbation factors, e.g. for speed or tempo perturbation.
     """
-
     def __init__(self, prob, minimum, maximum):
         self.prob = prob
         self.min = minimum
         self.max = maximum
 
-
 class WaveformPerturbation:
     """
-    This class enables the perturbation of audio waveforms by applying a variety of transformations such as speed and
-    tempo modification, SoX effects, codec application, and pre-emphasis filtering.
+    This class enables the perturbation of audio waveforms by applying a variety of transformations such
+    as speed and tempo modification, SoX effects, and pre-emphasis filtering.
     The parameters `speed`, `tempo`, `codecs`, and `preemphasis` contain a 'prob' key
-    which determines the probability that the corresponding transformation is applied. 
+    which determines the probability that the corresponding transformation is applied.
     """
 
     def __init__(
         self,
         speed: Optional[Dict[str, Any]] = None,
         tempo: Optional[Dict[str, Any]] = None,
         sox_effects: Optional[List[List[str]]] = None,
-        codecs: Optional[List[Dict[str, Any]]] = None,
```

Comment:
Maybe it would be better to not completely remove the codec stuff but just raise a `NotImplementedError` if they are called for now because we probably want to add this lateron.

Reply:
Done

Reply:
Now you still remove it here, so the codec function can never be called.